app/models/generic_object.rb

```diff
@@ -0,0 +1,103 @@
+class GenericObject < ApplicationRecord
+  validates :name, :presence => true, :uniqueness => true
+
+  belongs_to :generic_object_definition
+  has_many   :custom_attributes, :as => :resource, :dependent => :destroy, :autosave => true
+
+  TYPE_MAP = {
+    :boolean     => ActiveRecord::Type::Boolean.new,
+    :datetime    => ActiveRecord::Type::DateTime.new,
+    :time        => ActiveRecord::Type::Time.new,
+    :float       => ActiveRecord::Type::Float.new,
+    :integer     => ActiveRecord::Type::Integer.new,
+    :string      => ActiveRecord::Type::String.new,
+  }
+
+  def initialize(klass, attributes = {})
+    definition_object(klass)
+    raise "Generic Object Definition is required for creating a Generic Object." unless @definition_object
+    attributes[:generic_object_definition] = @definition_object
+    super(attributes)
+  end
+
+  def definition_object(klass = nil)
+    @definition_object ||= klass ? GenericObjectDefinition.find_by(:name => klass) : generic_object_definition
+  end
+
+  def type_cast(attr_name, value)
+    TYPE_MAP.fetch(attribute_type(attr_name)).cast(value)
+  end
+
+  def attribute_type(name)
+    defined_attributes[name].to_sym
+  end
+
+  def defined_attributes
+    @defined_attributes ||=
+      definition_object.properties[:attributes].each_with_object({}) { |(attr, type), h| h[attr.to_s] = type }
+  end
+
+  def inspect
+    attributes_as_string = self.class.column_names.collect do |name|
+      "#{name}: #{attribute_for_inspect(name)}"
+    end
+
+    attributes_as_string += custom_attributes(true).collect do |ca|
+      value = type_cast(ca.name, ca.value)
+      "#{ca.name}: #{ca_attribute_for_inspect(value)}"
+    end
+
+    "#<#{self.class} #{attributes_as_string.compact.join(", ")}>"
+  end
+
+  def method_missing(method_name, *args)
+    _log.info("  method_name = #{method_name}")
+    m = method_name.to_s
+    m.chop! if m.ends_with?('=')
```

Comment:
This can simplify to `m.chomp!("=")`  chomp will remove the `=` at the end and if it doesn't exist, it will do nothing, so it avoids the end_with? call